mParticle-Appboy/MPKitAppboy.m

```diff
@@ -192,9 +191,15 @@ - (NSString *)advertisingIdentifierString {
 
 #pragma mark ABKAppboyEndpointDelegate
 - (NSString *)getApiEndpoint:(NSString *)appboyApiEndpoint {
-    NSString *host = self.host;
-    NSString *endpoint = [appboyApiEndpoint stringByReplacingOccurrencesOfString:originalHost withString:host];
-    return endpoint;
+    NSMutableString *modifiedEndpoint = [appboyApiEndpoint mutableCopy];
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:
+                                  @"https.*\\.com" options:0 error:nil];
+    
+    if (self.host) {
+        [regex replaceMatchesInString:modifiedEndpoint options:0 range:NSMakeRange(0, [modifiedEndpoint length]) withTemplate:self.host];
+    }
+    
+    return modifiedEndpoint;
```

Comment:
Wouldn't have bothered to mention this if I didn't have the other feedback but likely worth adding ` copy]` here to avoid exposing the internal mutable object